libethereum/BlockQueue.h

```diff
@@ -67,6 +67,156 @@ enum class QueueStatus
 
 std::ostream& operator<< (std::ostream& os, QueueStatus const& obj);
 
+template<class T>
+class SizedBlockQueue
+{
+public:
+	std::size_t count() const { return m_queue.size(); }
+
+	std::size_t size() const { return m_size; }
+	
+	bool isEmpty() const { return m_queue.empty(); }
```

Comment:
It is more in accordance with our coding standards for bool getters, it is not a standard container and `empty()` is ambiguous as can be interpreted as "empty it"